packages/contracts/src/contracts/current/protocol/Exchange/MixinWrapperFunctions.sol

```diff
@@ -16,281 +16,259 @@
-        //
-        // | Offset | Length | Contents                     |
-        // |--------|--------|------------------------------|
-        // | 0      | 4      | function selector            |
-        // | 4      | 160    | address[5] orderAddresses    |
-        // | 164    | 192    | uint256[6] orderValues       |
-        // | 356    | 32     | uint256 takerTokenFillAmount |
-        // | 388    | 32     | offset to signature (416)    |
-        // | 420    | 32     | len(signature)               |
-        // | 452    | (1)    | signature                    |
-        // | (2)    | (3)    | padding (zero)               |
-        // | (4)    |        | end of input                 |
-        //
-        // (1): len(signature)
-        // (2): 452 + len(signature)
-        // (3): (32 - len(signature)) mod 32
-        // (4): 452 + len(signature) + (32 - len(signature)) mod 32
-        //
-        // [1]: https://solidity.readthedocs.io/en/develop/abi-spec.html
         
-        // Allocate memory for input
-        uint256 signatureLength = signature.length;
-        uint256 paddingLength = (32 - signatureLength) % 32;
-        uint256 inputSize = 452 + signatureLength + paddingLength;
-        bytes memory input = new bytes(inputSize);
+        // | Offset | Length  | Contents                     |
+        // |--------|---------|------------------------------|
+        // | 0      | 4       | function selector            |
+        // | 4      | 11 * 32 | Order order                  |
+        // | 356    | 32      | uint256 takerTokenFillAmount |
+        // | 388    | 32      | offset to signature (416)    |
+        // | 420    | 32      | len(signature)               |
+        // | 452    | (1)     | signature                    |
+        // | (2)    | (3)     | padding (zero)               |
+        // | (4)    |         | end of input                 |
         
-        // The in memory layout of `bytes memory input` starts with
-        // `uint256 length`, the content of the byte array starts at
-        // offset 32 from the pointer. We need assembly to access the
-        // raw pointer value of `input`.
-        // TODO: I can not find an official source for this.
-        uint256 start;
-        assembly {
-            start := add(input, 32)
-        }
-        
-        // Write function signature
-        // We use assembly to write four bytes at a time (actually 32,
-        // but we are only overwriting uninitialized data). A `bytes4`
-        // is stored by value as 256-bit and right-padded with zeros.
-        bytes4 FILL_ORDER_FUNCTION_SIGNATURE = bytes4(keccak256("fillOrder(address[5],uint256[6],uint256,uint8,bytes32,bytes32)"));
-        assembly {
-            mstore(start, FILL_ORDER_FUNCTION_SIGNATURE)
-        }
+        // (1): len(signature)
+        // (2): 452 + len(signature)
+        // (3): 32 - (len(signature) mod 32)
```

Comment:
When `len(signature)` is an exact multiple of 32, this incorrectly pads by 32 bytes, where 0 bytes are required.

A correct formula is `(32 - len(signature)) mod 32`. This can be simplified to `(-len(signature)) mod 32`.